presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java

```diff
@@ -285,8 +286,8 @@ private PlanNodeCostEstimate calculateCumulativeCost(
                 planNode -> requireNonNull(stats.apply(planNode), "no stats for node"),
                 noLookup(),
                 session,
-                types.entrySet().stream()
-                        .collect(ImmutableMap.toImmutableMap(entry -> new Symbol(entry.getKey()), Map.Entry::getValue)));
+                TypeProvider.copyOf(types.entrySet().stream()
```

Comment:
TypeProvider could provide a collector ;)

Reply:
I don't think it's worth it. It's a one-off use case and TypeProvider will eventually be removed (see comment above).